encode.go

```diff
@@ -148,48 +149,56 @@ func newValueEncoder(t reflect.Type) valueEncoder {
 }
 
 func structEncoder(v reflect.Value) ([]byte, error) {
-	var specs []fieldSpec
-	for i := 0; i < v.Type().NumField(); i++ {
-		f := v.Type().Field(i)
-		var (
-			err  error
-			spec fieldSpec
-			ok   bool
-		)
-		spec.startPos, spec.endPos, ok = parseTag(f.Tag.Get("fixed"))
-		if !ok {
+	ss := cachedStructSpec(v.Type())
+	dst := bytes.Repeat([]byte(" "), ss.ll)
+
+	for i, spec := range ss.fieldSpecs {
+		if !spec.ok {
 			continue
 		}
-		spec.value, err = newValueEncoder(f.Type)(v.Field(i))
+
+		val, err := newValueEncoder(v.Field(i).Type())(v.Field(i))
 		if err != nil {
 			return nil, err
 		}
-		specs = append(specs, spec)
+		copy(dst[spec.startPos-1:spec.endPos:spec.endPos], val)
```

Comment:
Hi @ianlopshire 

Sorry for commenting in the closed pull request. But I have a question:
```
copy(dst[spec.startPos-1:spec.endPos:spec.endPos], val)
// and 
copy(dst[spec.startPos-1:spec.endPos], val)
```
Are they different?

Reply:
@huydang284, I believe they are equivalent in this context. I'm not sure if the added index affects performance. Feel free to benchmark it and submit a PR if it does.

Reply:
Hi @ianlopshire, thanks for your information.